components/google_drive/sources/new-files/new-files.js

```diff
@@ -0,0 +1,227 @@
+        `Checking for resubscription on resource ${subscription.resourceId}`
+      );
+      // If the subscription for this resource will expire before the next run,
+      // stop the existing subscription and renew. Expiration is in ms.
+      if (
+        subscription.expiration <
+        +new Date() + event.interval_seconds * 1000
+      ) {
+        console.log(
+          `Notifications for resource ${subscription.resourceId} are expiring at ${subscription.expiration}. Renewing`
+        );
+        await this.googleDrive.stopNotifications(
+          channelID,
+          subscription.resourceId
+        );
+        const { expiration, resourceId } = await this.googleDrive.watchDrive(
+          channelID,
+          this.http.endpoint,
+          pageToken,
+          this.drive === "myDrive" ? null : this.drive
+        );
+        this.db.set("subscription", { expiration, resourceId });
+      }
+      return;
+    }
+
+    const { headers } = event;
+
+    if (headers["x-goog-resource-state"] === "sync") {
+      console.log("Sync notification, exiting early");
+      return;
+    }
+
+    if (
+      !headers["x-goog-resource-state"] ||
+      !headers["x-goog-resource-id"] ||
+      !headers["x-goog-resource-uri"] ||
+      !headers["x-goog-message-number"]
+    ) {
+      console.log("Request missing necessary headers: ", headers);
+      return;
+    }
+
+    const incomingChannelID = headers["x-goog-channel-id"];
+    if (incomingChannelID !== channelID) {
+      console.log(
+        `Channel ID of ${incomingChannelID} not equal to deployed component channel of ${channelID}`
+      );
+      return;
+    }
+
+    if (headers["x-goog-resource-id"] !== subscription.resourceId) {
+      console.log(
+        `Resource ID of ${resourceId} not currently being tracked. Exiting`
+      );
+      return;
+    }
+
+    // We observed false positives where a single change to a document would trigger two changes:
+    // one to "properties" and another to "content,properties". But changes to properties
+    // alone are legitimate, most users just won't want this source to emit in those cases.
+    // If x-goog-changed is _only_ set to "properties", only move on if the user set the prop
+    if (
+      !this.watchForPropertiesChanges &&
+      headers["x-goog-changed"] === "properties"
+    ) {
+      console.log(
+        "Change to properties only, which this component is set to ignore. Exiting"
+      );
+      return;
+    }
+
+    const {
+      changedFiles,
+      newStartPageToken,
+    } = await this.googleDrive.getChanges(
+      pageToken,
+      this.drive === "myDrive" ? null : this.drive
+    );
+
+    this.db.set("pageToken", newStartPageToken);
+
+    // only get files created since the last logged file creation or within the last five minutes
+    const fiveMinutesAgo = new Date(Date.now() - 5 * 60000);
+    const lastFileCreatedTime =
+      new Date(this.db.get("lastFileCreatedTime")) || fiveMinutesAgo;
```

Comment:
That's to limit the files that would be emitted on the first run in case the user has a large amount of files in their drive. Five minutes isn't very long though. Would you suggest maybe a week or a month instead?

Reply:
Oh I see your point. In that case, setting the date to `Date.now()` should be the right approach (i.e. from the moment the event source was activated).

We need to be careful not to override it during configuration changes (which trigger the `activate`/`deactivate` hooks):
```js
module.exports = {
  hooks: {
    activate() {
      const lastFileCreatedTime = this.db.get("lastFileCreatedTime") || Date.now();
      this.db.set("lastFileCreatedTime", lastFileCreatedTime);
    },
  },
};
```

Anyway, your solution will work but it might pull old files. It's minor, so I won't block the PR because of this, we can change it later.